source/common/router/config_impl.h

```diff
@@ -1076,12 +1137,7 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
   const Http::LowerCaseString cluster_header_name_;
   ClusterSpecifierPluginSharedPtr cluster_specifier_plugin_;
   const std::chrono::milliseconds timeout_;
-  const absl::optional<std::chrono::milliseconds> idle_timeout_;
-  const absl::optional<std::chrono::milliseconds> max_stream_duration_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_header_max_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_header_offset_;
-  const absl::optional<std::chrono::milliseconds> max_grpc_timeout_;
-  const absl::optional<std::chrono::milliseconds> grpc_timeout_offset_;
+  std::unique_ptr<const OptionalTimeouts> optional_timeouts_;
```

Comment:
const

Reply:
The common patter here is `unique_ptr<const T>`, this is consistent with the rest.

Reply:
I would suggest to not lose the constness on these timeouts. I dont see the pattern or what it brings us.